var/spack/repos/builtin/packages/py-numpy-quaternion/package.py

```diff
@@ -17,21 +17,28 @@ class PyNumpyQuaternion(PythonPackage):
     C for speed."""
 
     homepage = "https://github.com/moble/quaternion"
-    pypi = "numpy-quaternion/numpy-quaternion-2021.11.4.15.26.3.tar.gz"
+    pypi = "numpy-quaternion/numpy_quaternion-2024.0.3.tar.gz"
 
     license("MIT")
 
+    version("2024.0.3", sha256="cf39a8a4506eeda297ca07a508c10c08b3487df851a0e34f070a7bf8fab9f290")
     version(
         "2021.11.4.15.26.3",
         sha256="b0dc670b2adc8ff2fb8d6105a48769873f68d6ccbe20af6a19e899b1e8d48aaf",
+        url="https://pypi.io/packages/source/n/numpy-quaternion/numpy-quaternion-2021.11.4.15.26.3.tar.gz",
     )
 
     depends_on("c", type="build")  # generated
 
     variant("scipy", default=True, description="Build with scipy support")
     variant("numba", default=True, description="Build with numba support")
 
+    depends_on("python@3.10:", when="@2024.0.3:")
+
     depends_on("py-setuptools", type="build")
     depends_on("py-numpy@1.13:", type=("build", "run"))
+    depends_on("py-numpy@1.25:2", type=("build", "run"), when="@2024.0.3:")
     depends_on("py-scipy", type=("build", "run"), when="+scipy")
+    depends_on("py-scipy@1.5:1", type=("build", "run"), when="@2024.0.3:+scipy")
     depends_on("py-numba", type=("build", "run"), when="+numba")
+    depends_on("py-numba@0.55:", type=("build", "run"), when="@2024.0.3:+numba")
```

Comment:
The newer version requirements apply as of all 2024 versions:
```suggestion
    depends_on("python@3.10:", when="@2024:")

    depends_on("py-setuptools", type="build")
    depends_on("py-numpy@1.13:", type=("build", "run"))
    depends_on("py-numpy@1.25:2", type=("build", "run"), when="@2024:")
    depends_on("py-scipy", type=("build", "run"), when="+scipy")
    depends_on("py-scipy@1.5:1", type=("build", "run"), when="@2024:+scipy")
    depends_on("py-numba", type=("build", "run"), when="+numba")
    depends_on("py-numba@0.55:", type=("build", "run"), when="@2024:+numba")
```
There is also
```diff
 [build-system]
-requires = ["setuptools!=50.0", "wheel", "oldest-supported-numpy"]
+requires = ["hatchling", "setuptools>=61", "wheel", "numpy>=2.0,<3"]
```
between [v2023.0.4...v2024.0.2](https://github.com/moble/quaternion/compare/v2023.0.4...v2024.0.2#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711), which is a bit inconsistent with the other bounds you have. Does this install without a dependency on hatchling and with a numpy < 2?

Reply:
It did install locally for me, but maybe I had some things cached? I didn't try to run this in an isolated container build...

Should be fixed now. I split the numpy dependency into the different build/run requirements, too.